Library/Formula/task.rb

```diff
@@ -17,14 +15,17 @@ class Task < Formula
 
   head do
     url "https://git.tasktools.org/scm/tm/task.git", :branch => "2.5.1", :shallow => false
-
-    depends_on "gnutls"
   end
 
+  option "without-sync", "Disable sync support"
+
   depends_on "cmake" => :build
+  depends_on "gnutls" if !build.without? "sync"
```

Comment:
Just do `:optional` here and make `without-gnutls` note that it disables sync support.